src/zenml/services/service.py

```diff
@@ -476,22 +456,49 @@ def stop(self, timeout: int = 0, force: bool = False) -> None:
                 the service status.
             force: if True, the service will be stopped even if it is not
                 currently running.
-
-        Raises:
-            RuntimeError: if the service cannot be stopped
         """
         with console.status(f"Stopping service '{self}'.\n"):
             self.admin_state = ServiceState.INACTIVE
             self.deprovision(force)
             if timeout > 0:
                 self.poll_service_status(timeout)
                 if not self.is_stopped:
-                    raise RuntimeError(
+                    logger.error(
                         f"Failed to stop service {self}. Last state: "
                         f"'{self.status.state.value}'. Last error: "
                         f"'{self.status.last_error}'"
                     )
 
+    def get_prediction_url(self) -> Optional[str]:
+        """Gets the prediction URL for the endpoint.
+
+        Returns:
+            the prediction URL for the endpoint
+        """
+        prediction_url = None
+        if isinstance(self, BaseDeploymentService) and self.prediction_url:
+            prediction_url = self.prediction_url
+        elif self.endpoint:
+            prediction_url = (
+                self.endpoint.status.uri if self.endpoint.status else None
+            )
+        return prediction_url
+
+    def get_healthcheck_url(self) -> Optional[str]:
```

Comment:
There is a `get_healthcheck_url` method and a `healthcheck_url` property. Should they be somewhat related to each other?

Reply:
the `healthcheck_url` is a property that can be set for the service itself, the `get_healthcheck_url` check the health check for the service or the endpoint and return the first one it finds defined